lib/ecto/query/builder.ex

```diff
@@ -215,6 +215,13 @@ defmodule Ecto.Query.Builder do
     {expr, params}
   end
 
+  def escape({op, _, _}, _type, _params, _vars, _env) when op in ~w(|| && !)a do
+    error! """
+          Short-form operators are not supported: `#{op}`
+          Instead use the full name as found in SQL: `and`, `or`, and `not`.
+          """
```

Comment:
Let's reword this a little bit, as suggested:
```elixir
error! "boolean operators, such as `#{op}`, are not supported by Ecto." <>
       "They can be replaced with: `and`, `or`, and `not`"
```

Let's also not use multiline string - it will be displayed broken up and usually we display the errors for the width of terminal.